src/Commands/core/ArchiveDumpCommands.php

```diff
@@ -166,9 +168,48 @@ private function createArchiveFile(array $archiveComponents, array $options): st
         $archivePath = Path::join(dirname($this->archiveDir), self::ARCHIVE_FILE_NAME);
 
         stream_wrapper_restore('phar');
+        $this->logger()->info(var_export($archivePath, true));
         $archive = new PharData($archivePath);
 
         $this->createManifestFile($options);
+
+        $this->logger()->info(var_export($this->archiveDir, true));
+
+        // If symlinks are disabled, convert symlinks to full content.
+        if (is_dir($this->archiveDir)) {
```

Comment:
Shouldn't archiveDir always be a directory? If creating it might fail, then we should fail-fast above rather than nesting everything in an 'if'.